src/task/Pipeline.ts

```diff
@@ -268,14 +270,16 @@ export class Pipeline {
           this.dependencies.push([dep, id]);
         } else if (dep.startsWith("^") && packageName) {
           // topo dep -> build: ['^build']
-          const depTask = dep.substr(1);
-
+          const [depTask, dependencySet] = dep.startsWith("^^") ?
+            [dep.substr(2), this.getTransitiveGraphDependencies(packageName)]
```

Comment:
This function seems like it can get expensive... the question is whether we can calculate these upfront (or maybe only have ONE graph traversal).

Reply:
Yeah, this could be precomputed for the whole graph. Or at least, deferred until the a '^^' dependency is actually requested.

Reply:
Pushed an update that caches the transitive dependencies on a package.

In doing so I made it a recursive function because computing multiple outputs and representing the intermediate state of "walking in progress" is much easier in much more simply represented as a recursive walk.

![image](https://user-images.githubusercontent.com/1174858/145311050-3ddc2310-e124-4451-b32c-07c035043e06.png)

Not sure how big each stackframe of this fn is, so not sure what the max walk depth would be. 

Unfortunately, it is also not tail-call optimizable because it fans out to multiple children during recursion.